appl/include/obj_parser.h

```diff
@@ -1,181 +1,567 @@
 #ifndef OBJ_PARSER
 #define OBJ_PARSER
 
+#define POSITION_LABEL "v "
+#define POSITION_LABEL_CHAR_AMOUNT 2
+#define NORMAL_LABEL "vn "
+#define NORMAL_LABEL_CHAR_AMOUNT 3
+#define UV_LABEL "vt "
+#define UV_LABEL_CHAR_AMOUNT 3
+#define FACE_LABEL "f "
+#define FACE_LABEL_CHAR_AMOUNT 2
+#define INFO_PER_TRIANGLE_AMOUNT 3
+#define VERTEX_PER_TRIANGLE 3
+#define STARTING_DYNAMIC_ARRAY_CAPACITY 10
+
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
 
-typedef struct obj_vertex_info_t
-{
-    int v_index;
-    int vt_index;
-    int vn_index;
-} obj_vertex_info_t;
-
+/// @brief Structure to represent a 3D floating-point vector
 typedef struct obj_float3_t
 {
     float x;
     float y;
     float z;
 } obj_float3_t;
 
+/// @brief Structure to represent a 2D floating-point vector
 typedef struct obj_float2_t
 {
     float x;
     float y;
 } obj_float2_t;
 
-typedef struct obj_t
+/// @brief Structure to represent a vertex with position, normal, and texture coordinates
+typedef struct obj_vertex_t
 {
-    int v_count;
-    int vt_count;
-    int vn_count;
-    int f_count;
-    obj_float3_t *v;
-    obj_float2_t *vt;
-    obj_float3_t *vn;
-    obj_vertex_info_t *v_info;
-
-} obj_t;
+  obj_float3_t position;
+  obj_float3_t normal;
+  obj_float2_t uv;
+} obj_vertex_t;
 
-obj_t *obj_parse(const char *file_name)
+/// @brief Structure to represent a triangle composed of three vertices
+typedef struct obj_triangle_t
 {
+  obj_vertex_t v1;
+  obj_vertex_t v2;
+  obj_vertex_t v3;
+} obj_triangle_t;
 
-    obj_t *obj = (obj_t *)malloc(sizeof(obj_t));
-    obj->v_count = 0;
-    obj->vt_count = 0;
-    obj->vn_count = 0;
-    obj->f_count = 0;
-
-    FILE *file;
-    fopen_s(&file, file_name, "r");
+/// @brief Structure to hold dynamic arrays of position, normal, texture coordinates, and vertex information
+typedef struct obj_info_t
+{
+    obj_float3_t* position_dynamic_array;
+    size_t position_amount;
+    size_t position_dynamic_array_capacity;
+    obj_float3_t* normal_dynamic_array;
+    size_t normal_amount;
+    size_t normal_dynamic_array_capacity;
+    obj_float2_t* uv_dynamic_array;
+    size_t uv_amount;
+    size_t uv_dynamic_array_capacity;
+    obj_vertex_t* vertex_dynamic_array;
+    size_t vertex_amount;
+    size_t vertex_dynamic_array_capacity;
+} obj_info_t;
+
+/// @brief Structure to hold dynamic arrays of triangles
+typedef struct obj_t
+{
+    obj_triangle_t* triangles;
+    size_t triangle_amount;
+    size_t triangles_capacity;
```

Comment:
as far the obj_t is the "object" that a final user obtain, the `triangles_capacity` is not an information that need to be exposed to the final user. So better to avoid to store it in the `obj_t`